### DISCUSSION TITLE
Why is the RegistryKey created repeatedly in FabricDimensionTest?

### DISCUSSION THREAD

**Discussion by xtexx** [FabricMC/fabric-api#1613] [Mod Dev Support] [answered]
In [FabricDimensionTest.java](https://github.com/FabricMC/fabric/blob/1.17/fabric-dimensions-v1/src/testmod/java/net/fabricmc/fabric/test/dimension/FabricDimensionTest.java)(Fabric Test Mod),

[Line 55](https://github.com/FabricMC/fabric/blob/1.17/fabric-dimensions-v1/src/testmod/java/net/fabricmc/fabric/test/dimension/FabricDimensionTest.java#L55): `private static RegistryKey<World> WORLD_KEY = RegistryKey.of(`
[Line 69](https://github.com/FabricMC/fabric/blob/1.17/fabric-dimensions-v1/src/testmod/java/net/fabricmc/fabric/test/dimension/FabricDimensionTest.java#L69)(`onInitialize`): `WORLD_KEY = RegistryKey.of(Registry.WORLD_KEY, new Identifier("fabric_dimension", "void"));`

They are all created an RegistryKey and setted to the same field, why? Can I remove the create in `onInitialize` in my mod?

**Comment by warjort** [comment, marked as answer]
Its probably something leftover from refactoring?
The code in onInitialize() looks redundant.

Test code doesn't usually have the same quality as production code. :-)

**Reply by xtexx** [reply]
Hmmm, yes. Thanks
